nixos/modules/programs/captive-browser.nix

```diff
@@ -81,21 +83,23 @@ in
 
   config = mkIf cfg.enable {
 
-    programs.captive-browser.dhcp-dns = mkOptionDefault (
+    programs.captive-browser.dhcp-dns = let
+      iface = prefix:
+        optionalString cfg.bindInterface (escapeShellArg (concatStringsSep " " [ prefix cfg.interface ]));
+    in mkOptionDefault (
       if config.networking.networkmanager.enable then
-        "${pkgs.networkmanager}/bin/nmcli dev show ${escapeShellArg cfg.interface} | ${pkgs.gnugrep}/bin/fgrep IP4.DNS"
+        "${pkgs.networkmanager}/bin/nmcli dev show ${iface ""} | ${pkgs.gnugrep}/bin/fgrep IP4.DNS"
       else if config.networking.dhcpcd.enable then
-        "${pkgs.dhcpcd}/bin/dhcpcd -U ${escapeShellArg cfg.interface} | ${pkgs.gnugrep}/bin/fgrep domain_name_servers"
+        "${pkgs.dhcpcd}/bin/dhcpcd ${iface "-U"} | ${pkgs.gnugrep}/bin/fgrep domain_name_servers"
```

Comment:
it will produce ```'-U eth0'``` quoting them both

actually, I would inline the `iface` function to make it more readable